test/common/testlib.py

```diff
@@ -1399,6 +1399,7 @@ def setUp(self, restrict=True):
                 raise unittest.SkipTest("Cannot provision machines if test is marked as nondestructive")
             self.machine = self.machines['machine1'] = MachineCase.get_global_machine()
         else:
+            MachineCase.kill_global_machine()
```

Comment:
The problem with this is that we're going to start running everything in a single Python process now.  If you run a test that uses the global machine, it gets set up.  Then when you run the next test which is not using the global machine it doesn't get torn down.  That means you have an idle machine in the background using resources.

I also agree that this is likely to cause problems vis-a-vis `run()`, and indeed, I didn't test this with direct invocations.  I'd be happy enough to drop this, because we sort the tests to run the global machine cases at the end, so in theory it ought not to make a big difference.

I suppose it's a bit of a theoretical question.  Choose your poison:
 - always force the tests to be sorted such that the global machine tests are guaranteed to be grouped together at the end; or
 - waste resources by tearing down the global machine while you run the non-global-machine cases, only to risk having to start the global machine again later; or
 - waste memory by allowing the idle global machine to continue running in the background while running cases that aren't using it